core/src/main/java/com/opensymphony/xwork2/util/AbstractLocalizedTextProvider.java

```diff
@@ -413,11 +411,11 @@ public ResourceBundle findResourceBundle(String aBundleName, Locale locale) {
                     }
                 } catch (MissingResourceException e) {
                     LOG.debug("Missing resource bundle [{}]!", aBundleName, e);
-                    missingBundles.add(key);
+                    missingBundles.put(key, Boolean.TRUE);
                 }
             } else {
                 LOG.debug("Missing resource bundle [{}]!", aBundleName);
-                missingBundles.add(key);
+                missingBundles.put(key, Boolean.TRUE);
```

Comment:
And the same `missingBundles.putIfAbsent(key, Boolean.TRUE);` consideration here.

Reply:
Same as above.